Add litmus book to install and remove storageclass for jiva and cstor for openebs v0.7

ashishranjan738 opened the pull request:
Signed-off-by: Ashish Ranjan <ashishranjan738@gmail.com>

<!--  Thanks for sending a pull request!  Here are some tips for you -->

**What this PR does / why we need it**:
-> This PR adds litmus book to install storage class for jiva and cstore both.
-> To make the book generic there is a environment variable to enable and disable applying for a specific engine.
-> Disk list for cstor storage pool claim can be provided through DISK_LIST environment variable in csv format.
-> Environment variables available are:
```
# Enable/Disable storageclass
-> APPLY_JIVA 
-> APPLY_CSTOR 
-> DELETE_JIVA  
-> DELETE_CSTOR  
# Jiva configuration  
-> JIVA_POOL_NAME  
-> JIVA_STORAGECLASS_NAME  
-> STORAGE_PATH  
-> OPENEBS_IO_JIVA_REPLICA_COUNT  
-> OPENEBS_IO_JIVA_CONTROLLER_IMAGE  
-> OPENEBS_IO_JIVA_REPLICA_IMAGE  
-> OPENEBS_IO_VOLUME_MONITOR_IMAGE  
# Cstor configuration  
-> CSTOR_POOL_NAME  
-> MAX_POOLS  
-> CSTOR_STORAGECLASS_NAME  
-> DISK_LIST  
```
**Special notes for your reviewer**:
-> This book has been tested on GCP .
-> logs of installation of cstore storage-class
```
ansible-playbook 2.6.2
  config file = None
  configured module search path = [u'/root/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/lib/python2.7/dist-packages/ansible
  executable location = /usr/local/bin/ansible-playbook
  python version = 2.7.15rc1 (default, Apr 15 2018, 21:51:34) [GCC 7.3.0]
No config file found; using defaults

PLAYBOOK: storageclass_setup.yaml **********************************************
1 plays in ./storageclass/0.7/ansible/storageclass_setup.yaml

PLAY [localhost] ***************************************************************
META: ran handlers

TASK [template] ****************************************************************
task path: /storageclass/0.7/ansible/storageclass_setup.yaml:42

TASK [Add the disks to the pool] ***********************************************
task path: /storageclass/0.7/ansible/storageclass_setup.yaml:46

TASK [Applying storage class] **************************************************
task path: /storageclass/0.7/ansible/storageclass_setup.yaml:53

TASK [Confirm that openebs storage pools are created] **************************
task path: /storageclass/0.7/ansible/storageclass_setup.yaml:58

TASK [Confirm that openebs storage classes are created] ************************
task path: /storageclass/0.7/ansible/storageclass_setup.yaml:67
META: ran handlers
META: ran handlers

PLAY RECAP *********************************************************************
127.0.0.1                  : ok=7    changed=5    unreachable=0    failed=0   

```
-> logs of removal of cstor storage-class 
```
ansible-playbook 2.6.2
  config file = None
  configured module search path = [u'/root/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/lib/python2.7/dist-packages/ansible
  executable location = /usr/local/bin/ansible-playbook
  python version = 2.7.15rc1 (default, Apr 15 2018, 21:51:34) [GCC 7.3.0]
No config file found; using defaults

PLAYBOOK: storageclass_cleanup.yaml ********************************************
1 plays in ./storageclass/0.7/ansible/storageclass_cleanup.yaml

PLAY [localhost] ***************************************************************
META: ran handlers

TASK [template] ****************************************************************
task path: /storageclass/0.7/ansible/storageclass_cleanup.yaml:13

TASK [Deleting storage class] **************************************************
task path: /storageclass/0.7/ansible/storageclass_cleanup.yaml:17
fatal: [127.0.0.1]: FAILED! => {"changed": true, "cmd": "kubectl delete -f sc_jiva.yaml", "delta": "0:00:01.310127", "end": "2018-08-15 04:14:05.178176", "msg": "non-zero return code", "rc": 1, "start": "2018-08-15 04:14:03.868049", "stderr": "Error from server (NotFound): error when deleting \"sc_jiva.yaml\": storagepools.openebs.io \"openebs-mntdir\" not found\nError from server (NotFound): error when deleting \"sc_jiva.yaml\": storageclasses.storage.k8s.io \"openebs-storageclass\" not found", "stderr_lines": ["Error from server (NotFound): error when deleting \"sc_jiva.yaml\": storagepools.openebs.io \"openebs-mntdir\" not found", "Error from server (NotFound): error when deleting \"sc_jiva.yaml\": storageclasses.storage.k8s.io \"openebs-storageclass\" not found"], "stdout": "", "stdout_lines": []}
...ignoring

TASK [Confirm that openebs storage pools are deleted] **************************
task path: /storageclass/0.7/ansible/storageclass_cleanup.yaml:23

TASK [Confirm that openebs storage classes are deleted] ************************
task path: /storageclass/0.7/ansible/storageclass_cleanup.yaml:32

TASK [template] ****************************************************************
task path: /storageclass/0.7/ansible/storageclass_cleanup.yaml:43

TASK [Add the disks to the pool] ***********************************************
task path: /storageclass/0.7/ansible/storageclass_cleanup.yaml:47

TASK [Deleting storage class] **************************************************
task path: /storageclass/0.7/ansible/storageclass_cleanup.yaml:54

TASK [Confirm that openebs storage pools are deleted] **************************
task path: /storageclass/0.7/ansible/storageclass_cleanup.yaml:60

TASK [Confirm that openebs storage classes are deleted] ************************
task path: /storageclass/0.7/ansible/storageclass_cleanup.yaml:69
META: ran handlers
META: ran handlers

PLAY RECAP *********************************************************************
127.0.0.1                  : ok=11   changed=9    unreachable=0    failed=0   
```

yudaykiran commented:
@ashishranjan738 , can you provide the output for running cstor and jiva installation and removal in more verbose mode?

ashishranjan738 commented:
yes sure sir

ashishranjan738 commented:
logs of jiva storageclass deletion:
```ansible-playbook 2.6.2
  config file = None
  configured module search path = [u'/root/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/lib/python2.7/dist-packages/ansible
  executable location = /usr/local/bin/ansible-playbook
  python version = 2.7.15rc1 (default, Apr 15 2018, 21:51:34) [GCC 7.3.0]
No config file found; using defaults

PLAYBOOK: storageclass_cleanup.yaml ********************************************
1 plays in ./storageclass/0.7/ansible/storageclass_cleanup.yaml
PLAY [localhost] ***************************************************************
TASK [Gathering Facts] *********************************************************
task path: /storageclass/0.7/ansible/storageclass_cleanup.yaml:4
ok: [127.0.0.1]
META: ran handlers
TASK [template] ****************************************************************
task path: /storageclass/0.7/ansible/storageclass_cleanup.yaml:13
changed: [127.0.0.1] => {"changed": true, "checksum": "ba43bcb7e30d31c54f7b425ef9a5bc6077272cef", "dest": "./sc_jiva.yaml", "gid": 0, "group": "root", "md5sum": "eb7fc04340779aaf81efceb58728786a", "
mode": "0644", "owner": "root", "size": 834, "src": "/root/.ansible/tmp/ansible-tmp-1534926919.56-214088789097830/source", "state": "file", "uid": 0}
TASK [Deleting storage class] **************************************************
task path: /storageclass/0.7/ansible/storageclass_cleanup.yaml:17
changed: [127.0.0.1] => {"changed": true, "cmd": "kubectl delete -f sc_jiva.yaml", "delta": "0:00:01.272619", "end": "2018-08-22 08:35:21.648673", "rc": 0, "start": "2018-08-22 08:35:20.376054", "st
derr": "", "stderr_lines": [], "stdout": "storagepool \"openebs-mntdir\" deleted\nstorageclass \"openebs-storageclass\" deleted", "stdout_lines": ["storagepool \"openebs-mntdir\" deleted", "storagec
lass \"openebs-storageclass\" deleted"]}
TASK [Confirm that openebs storage pools are deleted] **************************
task path: /storageclass/0.7/ansible/storageclass_cleanup.yaml:23
changed: [127.0.0.1] => {"attempts": 1, "changed": true, "cmd": "kubectl get storagepool", "delta": "0:00:00.920625", "end": "2018-08-22 08:35:22.773285", "rc": 0, "start": "2018-08-22 08:35:21.8526
60", "stderr": "", "stderr_lines": [], "stdout": "NAME      AGE\ndefault   12m", "stdout_lines": ["NAME      AGE", "default   12m"]}
TASK [Confirm that openebs storage classes are deleted] ************************
task path: /storageclass/0.7/ansible/storageclass_cleanup.yaml:32
changed: [127.0.0.1] => {"attempts": 1, "changed": true, "cmd": "kubectl get storageclass", "delta": "0:00:00.903325", "end": "2018-08-22 08:35:23.889657", "rc": 0, "start": "2018-08-22 08:35:22.986
332", "stderr": "", "stderr_lines": [], "stdout": "NAME                        PROVISIONER\nopenebs-snapshot-promoter   volumesnapshot.external-storage.k8s.io/snapshot-promoter\nopenebs-standard    
        openebs.io/provisioner-iscsi\nstandard (default)          kubernetes.io/gce-pd", "stdout_lines": ["NAME                        PROVISIONER", "openebs-snapshot-promoter   volumesnapshot.exter
nal-storage.k8s.io/snapshot-promoter", "openebs-standard            openebs.io/provisioner-iscsi", "standard (default)          kubernetes.io/gce-pd"]}
TASK [template] ****************************************************************
task path: /storageclass/0.7/ansible/storageclass_cleanup.yaml:43
skipping: [127.0.0.1] => {"changed": false, "skip_reason": "Conditional result was False"}
TASK [Add the disks to the pool] ***********************************************
task path: /storageclass/0.7/ansible/storageclass_cleanup.yaml:47
skipping: [127.0.0.1] => (item=)  => {"changed": false, "item": "", "skip_reason": "Conditional result was False"}
TASK [Deleting storage class] **************************************************
task path: /storageclass/0.7/ansible/storageclass_cleanup.yaml:54
skipping: [127.0.0.1] => {"changed": false, "skip_reason": "Conditional result was False"}
TASK [Confirm that openebs storage pools are deleted] **************************
task path: /storageclass/0.7/ansible/storageclass_cleanup.yaml:60
skipping: [127.0.0.1] => {"changed": false, "skip_reason": "Conditional result was False"}
TASK [Confirm that openebs storage classes are deleted] ************************
task path: /storageclass/0.7/ansible/storageclass_cleanup.yaml:69
skipping: [127.0.0.1] => {"changed": false, "skip_reason": "Conditional result was False"}
TASK [set_fact] ****************************************************************
task path: /storageclass/0.7/ansible/storageclass_cleanup.yaml:79
ok: [127.0.0.1] => {"ansible_facts": {"flag": "Pass"}, "changed": false}
META: ran handlers
META: ran handlers
PLAY RECAP *********************************************************************
127.0.0.1                  : ok=6    changed=4    unreachable=0    failed=0
